spyder/plugins/switcher/widgets/switcher.py

```diff
@@ -0,0 +1,472 @@
+      SwitcherSeparator: [---------------------------------------]
+      SwitcherItem:      [title description    <shortcut> section]
+      SwitcherItem:      [title description    <shortcut> section]
+    """
+
+    # Dismissed switcher
+    sig_rejected = Signal()
+    # Search/Filter text changes
+    sig_text_changed = Signal(TEXT_TYPES[-1])
+    # Current item changed
+    sig_item_changed = Signal(object)
+    # List item selected, mode and cleaned search text
+    sig_item_selected = Signal(object, TEXT_TYPES[-1], TEXT_TYPES[-1], )
+    sig_mode_selected = Signal(TEXT_TYPES[-1])
+
+    _MAX_NUM_ITEMS = 15
+    _MIN_WIDTH = 580
+    _MIN_HEIGHT = 200
+    _MAX_HEIGHT = 390
+    _ITEM_WIDTH = _MIN_WIDTH - 20
+
+    def __init__(self, parent, help_text=None, item_styles=ITEM_STYLES,
+                 item_separator_styles=ITEM_SEPARATOR_STYLES):
+        """Multi purpose switcher."""
+        super(Switcher, self).__init__(parent)
+        self._modes = {}
+        self._mode_on = ''
+        self._item_styles = item_styles
+        self._item_separator_styles = item_separator_styles
+
+        # Widgets
+        self.edit = QLineEdit(self)
+        self.list = QListView(self)
+        self.model = QStandardItemModel(self.list)
+        self.proxy = SwitcherProxyModel(self.list)
+        self.filter = KeyPressFilter()
+
+        # Widgets setup
+        self.setWindowFlags(Qt.Popup | Qt.FramelessWindowHint)
+        self.setWindowOpacity(0.95)
+#        self.setMinimumHeight(self._MIN_HEIGHT)
+        self.setMaximumHeight(self._MAX_HEIGHT)
+        self.edit.installEventFilter(self.filter)
+        self.edit.setPlaceholderText(help_text if help_text else '')
+        self.list.setMinimumWidth(self._MIN_WIDTH)
+        self.list.setItemDelegate(SwitcherDelegate(self))
+        self.list.setFocusPolicy(Qt.NoFocus)
+        self.list.setSelectionBehavior(self.list.SelectItems)
+        self.list.setSelectionMode(self.list.SingleSelection)
+        self.list.setVerticalScrollMode(QAbstractItemView.ScrollPerItem)
+        self.proxy.setSourceModel(self.model)
+        self.list.setModel(self.proxy)
+
+        # Layout
+        layout = QVBoxLayout()
+        layout.addWidget(self.edit)
+        layout.addWidget(self.list)
+        self.setLayout(layout)
+
+        # Signals
+        self.filter.sig_up_key_pressed.connect(self.previous_row)
+        self.filter.sig_down_key_pressed.connect(self.next_row)
+        self.filter.sig_enter_key_pressed.connect(self.enter)
+        self.edit.textChanged.connect(self.setup)
+        self.edit.textChanged.connect(self.sig_text_changed)
+        self.edit.returnPressed.connect(self.enter)
+        self.list.clicked.connect(self.enter)
+        self.list.clicked.connect(self.edit.setFocus)
+        self.list.selectionModel().currentChanged.connect(
+            self.current_item_changed)
+        self.edit.setFocus()
+
+    # --- Helper methods
```

Comment:
```suggestion
    # ---- Helper methods
```

This small change makes this comment to show up in Spyder's Outline pane.